ethcore/src/client/client.rs

```diff
@@ -907,7 +907,7 @@ impl Client {
 	pub fn state_at(&self, id: BlockId) -> Option<State<StateDB>> {
 		// fast path for latest state.
 		match id.clone() {
-			BlockId::Pending => return self.miner.pending_state().or_else(|| Some(self.state())),
+			BlockId::Pending => return self.miner.pending_state(self.chain.read().best_block_number()).or_else(|| Some(self.state())),
```

Comment:
The issue is that the pending state might be stalled, so we check that by comparing block numbers.
We had a similar thing earlier (at least for RPC), but the behaviour was very inconsistent - some stuff was going via the client and some via RPC.